packages/react/src/merge-props/mergeProps.test.ts

```diff
@@ -199,7 +199,12 @@ describe('mergeProps', () => {
 
   describe('props getters', () => {
     it('calls the props getter with the props defined after it', () => {
-      const propsGetter = spy((props) => props);
+      let observedProps;
+      const propsGetter = spy((props) => {
+        observedProps = { ...props };
+        return props;
+      });
+
```

Comment:
`mergeProps` now uses a single accumulator props object and progressively mutates it in-place as it merges a new bundle of props. Which is fine, except that prop-getters see that object while they run, so if they keep a reference to that props object and later use it like these tests do (in other words, if prop getters are impure), then they'll see the final merged props instead of what they saw when the function was called.

`mergeProps` could fall back to copying props before passing them to a prop-getter, but that would make it slower. What do we think? Do we want to support prop-getters that keep a ref to props they see, or are we ok with this optimization?